commands.js

```diff
@@ -2160,6 +2152,146 @@ exports.commands = {
 					"/banword delete [words] - Removes the comma-separated list of phrases from the banword list. Requires: # & ~",
 					"/banword list - Shows the list of banned words in the current room. Requires: @ * # & ~"],
 
+	bl: 'blacklist',
+	blacklist: {
+		addname: 'add',
+		add: function (target, room, user, connection, cmd) {
+			if (!target) return this.parse('/help blacklist');
+			if (!user.can('declare', null, room)) return;
+			if (!room.chatRoomData) return this.errorReply("This room does not support blacklists.");
+			let targets = target.split(',').map(param => toId(param));
+
+			for (let i = 0; i < targets.length; i++) {
+				let targetUser = Users(targets[i]);
+				let name = (targetUser ? targetUser.name : targets[i]);
+				let userid = toId(name);
+
+				if (!userid) return this.errorReply("Invalid name: " + name + ".");
+				if (!targetUser && cmd === 'add') return this.errorReply("Cannot add offline user " + name + " to blacklist, use '/bl addname' instead.");
+
+				let entry = Punishments.roombannedUserids.get(room.id + '|' + userid);
```

Comment:
Not to mention this is wrong, it needs the expiration check; there's a reason we don't directly read from `Punishments.userids` ever